server/src/main/java/org/elasticsearch/index/query/SearchExecutionContext.java

```diff
@@ -177,10 +182,43 @@ public SearchExecutionContext(SearchExecutionContext source) {
         );
     }
 
+    /**
+     * Factory method generating a dummy context with partial functionality. Strictly test-only.
+     */
+    public static SearchExecutionContext newDummyForTests(
```

Comment:
This is test code, but gets included in the release. Can we move it to test/framework/main/java ? Maybe as factory method in a class in org.elasticsearch.search package in the test framework?